engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/XenserverSnapshotStrategy.java

```diff
@@ -279,7 +282,52 @@ public boolean deleteSnapshot(Long snapshotId) {
 
     @Override
     public boolean revertSnapshot(SnapshotInfo snapshot) {
-        throw new CloudRuntimeException("revert Snapshot is not supported");
+        if (canHandle(snapshot,SnapshotOperation.REVERT) == StrategyPriority.CANT_HANDLE) {
```

Comment:
What about changing this "canHandle" method to return a boolean if the snapshot can be handled, instead of a enum value?